### ISSUE TITLE
Scan-build test is not running on macOS

### ISSUE THREAD

**Issue by vikman90** [wazuh/wazuh#23176]
|Wazuh version|Reference|
|---|--|
| 4.9.0 (dev) | #20448 |

Tests are failing with this message:

```
/Users/runner/work/_temp/57bd2aa5-8d5f-4e29-be8b-955f376374a2.sh: line 3: /usr/local/opt/llvm@14/bin/scan-build: No such file or directory
```

Duplicate of #23148.

**Comment by vikman90** [comment]
## Build issues

- `make TARGET=agent` is trying to rebuild some pre-compiled libraries.
- _libplist_ is not pre-compiled. The build is failing due to missing build dependencies.

**Comment by vikman90** [comment]
We should backport this fix to 4.7.5.
